internal/controller/dnsrecord_healthchecks.go

```diff
@@ -119,7 +116,7 @@ func (r *DNSRecordReconciler) reconcileHealthCheckStatus(results []provider.Heal
 	dnsRecord.Status.HealthCheck.Probes = []v1alpha1.HealthCheckStatusProbe{}
 
 	for _, result := range results {
-		if result.ID == "" {
+		if result.Host == "" {
 			continue
 		}
```

Comment:
This seems like it was originally for a completely different purpose. Do you not need to continue now if ID == ""?